drivers/counter/Kconfig.gd32

```diff
@@ -6,7 +6,7 @@
 config COUNTER_TIMER_GD32
 	bool "GD32 timer counter driver"
 	default y
-	depends on DT_HAS_GD_GD32_TIMER_ENABLED && SOC_FAMILY_GD32_ARM
+	depends on DT_HAS_GD_GD32_TIMER_ENABLED
```

Comment:
Is this compatible with RISC-V ?

CC: @soburi

Reply:
The timer has not yet worked with GD32V.
Please keep the restriction for ARM only.